iroh/src/client.rs

```diff
@@ -624,6 +626,47 @@ where
         Ok(())
     }
 
+    /// Add an entry from a file path
+    pub async fn import_file(
+        &self,
+        author: AuthorId,
+        path: impl AsRef<Path>,
+        path_to_key: impl Fn(PathBuf) -> Bytes + Send + Sync + 'static,
+    ) -> Result<DocImportFileProgress> {
+        self.ensure_open()?;
+        let path = path.as_ref().to_path_buf();
+        let key = path_to_key(path.clone());
+        let stream = self
+            .0
+            .rpc
+            .server_streaming(DocImportFileRequest {
+                doc_id: self.id(),
+                author_id: author,
+                path,
+                key,
+            })
+            .await?;
+        Ok(DocImportFileProgress::new(stream))
+    }
+
+    /// Export an entry as a file to a given path.
+    pub async fn export_file(
```

Comment:
We should think about potential consequences that `import_file` runs in the node and `export_file` in the client. E.g. they could potentially run under different unix system users or permissions, and it might be unexpected for `import_file` to work for a path but `export_file` not (if e.g. the user running the console has less permissions than the node).

Not saying this needs to change, but something to think about. Maybe it does make sense to move `export_file` to the node as well, could also help with perf at least for big files to not have to copy them over the rpc conn.

Reply:
fully agree.  this iteration was my "get it done and ask for help" version. If we change the params to remove the `query` and `key_to_path` method, in favor of `key` and `path` (as was recommended to me on my call this morning w/ dig and rudi), I can then use the rpc `server_streaming` patterns that we already have set up!

appreciate the thoughtful feedback